### PR TITLE
[nvidia]:Patch to update nvidia plugin for GPU info

### PR DESCRIPTION
**Pull request by mamatha4**
This patch is to update nvidia plugin to collect
logs for Nvidia GPUs

Signed-off-by: Mamatha Inamdar <mamatha4@linux.vnet.ibm.com>
Reported-by: Borislav Stoymirski <borislav.stoymirski@bg.ibm.com>
Reported-by: Yesenia Jimenez <yesenia@us.ibm.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x ] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x ] Is the subject and message clear and concise?
- [x ] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x ] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/nvidia.py [hunk 1]
```diff
@@ -23,13 +23,22 @@ def setup(self):
...
         ]
 
         self.add_cmd_output(["nvidia-smi %s" % cmd for cmd in subcmds])
 
         query = ('gpu_name,gpu_bus_id,vbios_version,temperature.gpu,'
-                 'utilization.gpu,memory.total,memory.free,memory.used')
+                 'utilization.gpu,memory.total,memory.free,memory.used'
+                 'clocks.applications.graphics,clocks.applications.memory')
+        querypages = ('timestamp,gpu_bus_id,gpu_serial,gpu_uuid,'
+                       'retired_pages.address,retired_pages.cause')
```
Comment (TurboTurtle) [inline sos/report/plugins/nvidia.py:39]: One space over-indented here, throwing a flake8 error.

File: sos/report/plugins/nvidia.py [hunk 2]
```diff
@@ -23,13 +23,22 @@ def setup(self):
...
+            '-q -d ECC',
+            'nvlink -s',
+            'nvlink -e'
         ]
 
         self.add_cmd_output(["nvidia-smi %s" % cmd for cmd in subcmds])
 
         query = ('gpu_name,gpu_bus_id,vbios_version,temperature.gpu,'
-                 'utilization.gpu,memory.total,memory.free,memory.used')
+                 'utilization.gpu,memory.total,memory.free,memory.used'
```
Comment (pmoravec) [inline sos/report/plugins/nvidia.py:36]: Missing comma after `memory.used` (before the quote mark)?

File: sos/report/plugins/nvidia.py [hunk 3]
```diff
@@ -23,13 +23,22 @@ def setup(self):
...
+                 'clocks.applications.graphics,clocks.applications.memory')
+        querypages = ('timestamp,gpu_bus_id,gpu_serial,gpu_uuid,'
+                       'retired_pages.address,retired_pages.cause')
         self.add_cmd_output("nvidia-smi --query-gpu=%s --format=csv" % query)
+        self.add_cmd_output("nvidia-smi --query-gpu=%s --format=csv" % query)
+        self.add_cmd_output("nvidia-smi ----query-retired-pages=%s --format=csv" % querypages)
```
Comment (TurboTurtle) [inline sos/report/plugins/nvidia.py:42]: Line is too long, causing a flake8 error, please wrap accordingly.

```python
self.add_cmd_output(
    "nvidia-smi ----query-retired-pages=%s --format=csv" % querypages
)
```

should work

File: sos/report/plugins/logs.py [hunk 4]
```diff
@@ -54,6 +54,7 @@ def setup(self):
         ])
 
         self.add_cmd_output("journalctl --disk-usage")
+        self.add_cmd_output("journalctl -b 0 -t nvidia-persistenced")
```
Comment (TurboTurtle) [inline sos/report/plugins/logs.py:57]: Wrong place for this, put this in the `nvidia` plugin as well, please. There's also an `add_journal()` method available which is the preferred way to get actual journals, e.g.:

```python
self.add_journal(boot=0, identifier='nvidia-persistenced')
```

**Review by TurboTurtle — COMMENTED** [review]
Overall the additions are fine to me - just need to move the journal collection and fix the linting issues.

File: sos/report/plugins/nvidia.py [hunk 5]
```diff
@@ -23,13 +23,22 @@ def setup(self):
...
         self.add_cmd_output(["nvidia-smi %s" % cmd for cmd in subcmds])
 
         query = ('gpu_name,gpu_bus_id,vbios_version,temperature.gpu,'
-                 'utilization.gpu,memory.total,memory.free,memory.used')
+                 'utilization.gpu,memory.total,memory.free,memory.used'
+                 'clocks.applications.graphics,clocks.applications.memory')
+        querypages = ('timestamp,gpu_bus_id,gpu_serial,gpu_uuid,'
+                       'retired_pages.address,retired_pages.cause')
         self.add_cmd_output("nvidia-smi --query-gpu=%s --format=csv" % query)
+        self.add_cmd_output("nvidia-smi --query-gpu=%s --format=csv" % query)
```
Comment (TurboTurtle) [inline sos/report/plugins/nvidia.py:41]: Just saw this duplication, was this intentional?

Comment (mamatha4) [reply sos/report/plugins/nvidia.py:41]: This was not intentional..copy paste error..Thanks

Comment (mamatha4) [reply sos/report/plugins/nvidia.py:41]: Thanks for the review, I have updated all your comments and sent new version

File: sos/report/plugins/nvidia.py [hunk 6]
```diff
@@ -23,13 +23,23 @@ def setup(self):
...
 
         query = ('gpu_name,gpu_bus_id,vbios_version,temperature.gpu,'
-                 'utilization.gpu,memory.total,memory.free,memory.used')
+                 'utilization.gpu,memory.total,memory.free,memory.used,'
+                 'clocks.applications.graphics,clocks.applications.memory')
+        querypages = ('timestamp,gpu_bus_id,gpu_serial,gpu_uuid,'
+                      'retired_pages.address,retired_pages.cause')
         self.add_cmd_output("nvidia-smi --query-gpu=%s --format=csv" % query)
+        self.add_cmd_output("nvidia-smi --query-retired-pages=%s"
+                            "--format=csv" % querypages)
```
[the remark this answers is not in the document]
Comment (mamatha4) [reply sos/report/plugins/nvidia.py:42]: ok

**Review by TurboTurtle — APPROVED** [review]

**Review by pmoravec — APPROVED** [review]

**Comment by mamatha4** [comment]
any updates on this..is this patch integrated in sosreport tool?

**Pull request merged**